site/reference/fn/eval/README.md

```diff
@@ -0,0 +1,153 @@
+---
+title: "`eval`"
+linkTitle: "eval"
+type: docs
+description: >
+  Execute function on resources
+---
+
+<!--mdtogo:Short
+    Execute function on resources
+-->
+
+`eval` executes a function on resources in a directory. Functions are packaged
+as container images or binary executables.
+
+If the function fails (i.e. exits with non-zero status code), `eval` will
+abort and the local filesystem is left intact.
+
+Refer to the [Imperative Function Execution] for detailed overview.
+
+### Synopsis
+
+<!--mdtogo:Long-->
+
+```shell
+kpt fn eval [DIR|-] [flags] [-- fn-args]
+```
+
+#### Args
+
+```shell
+DIR|-:
+  Path to the local directory containing resources. Defaults to the current
+  working directory. Using '-' as the directory path will cause `eval` to
+  read resources from `stdin` and write the output to `stdout`.
+```
+
+#### Flags
+
+```shell
+--as-current-user:
+  Use the `uid` and `gid` of the kpt process for container function execution.
+  Container functions are executed as `nobody` user which has very limited
+  privileges. You may want to use this flag to run higher privilege operations
+  such as mounting the local filesystem.
+
+--dry-run:
```

Comment:
Why is this needed?

You can just do:

`eval - | <some program>`

Reply:
So I see only a narrow use-case for this that is to `execute this function, but don't have any side effects of writing to local filesystem, just show me the output`. It's like `kubectl apply --dry-run` enables you to run apply without apply the resource to cluster.

So didn't take it out. We can rationalize this as part of `out-of-place` support.

Reply:
I'd say remove this for now, unless you want to have render have dry-run as well.